router.c

```diff
@@ -1079,19 +1120,13 @@ static void router_round(router_t *router, router_request_t *req, uint8_t round)
                                 vj_index, timetext(time));
                 #endif
 
-                /* Target pruning, section 3.1 of RAPTOR paper. */
-                if ((router->best_time[router->target] != UNREACHED) &&
-                    (req->arrive_by ? time < router->best_time[router->target]
-                                    : time > router->best_time[router->target])) {
-                    #ifdef RRRR_DEBUG_VEHICLE_JOURNEY
-                    fprintf(stderr, "    (target pruning)\n");
-                    #endif
-
-                    /* We cannot break out of this journey_pattern entirely,
-                     * because re-boarding may occur at a later stop.
-                     */
-                    continue;
+                if (router->n_targets > 0) {
+                    /* Target pruning, section 3.1 of RAPTOR paper. */
+                    if (target_pruning (router, req, time)) {
+                        continue;
+                    }
                 }
```

Comment:
Mmh, i would not have done it like this: this way you have to check all the target's (aka a lot of work) for each reacahble journey-pattern-point. 
Since target-pruning is also handled within the function board_vehicle_journeys_within_days, there is no need to do it like this.